src/components/SignerPanel/ConfirmMsgSign.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { isElectron, noop } from '../../utils'
+
+import { AppType } from '../../prop-types'
+import { NoWeb3Provider, AccountLocked } from './Web3Errors'
+import { ImpossibleContent } from './ImpossibleContent'
+import SignMsgContent from './SignMsgContent'
+
+class ConfirmMsgSign extends React.Component {
+  static propTypes = {
+    account: PropTypes.string,
+    apps: PropTypes.arrayOf(AppType).isRequired,
+    hasAccount: PropTypes.bool.isRequired,
+    hasWeb3: PropTypes.bool.isRequired,
+    intent: PropTypes.object,
+    onClose: PropTypes.func.isRequired,
+    onRequestEnable: PropTypes.func,
+    onSign: PropTypes.func.isRequired,
+    signError: PropTypes.string,
+    signingEnabled: PropTypes.bool.isRequired,
+    walletProviderId: PropTypes.string.isRequired,
+  }
+
+  static defaultProps = {
+    account: '',
+    intent: {},
+    onRequestEnable: noop,
+  }
+  render() {
+    const {
+      account,
+      apps,
+      intent,
+      hasAccount,
+      hasWeb3,
+      onClose,
+      onRequestEnable,
+      onSign,
+      signError,
+      signingEnabled,
+      walletProviderId,
+    } = this.props
+
+    if (!hasWeb3) {
```

Comment:
You'll notice I added a component to make this check. The parent component was already very hefty (a conversation we had in the [previous version of this PR](https://github.com/aragon/aragon/pull/683)), and doing so would (seemingly) require some nested ternaries. 

Alternatively, this added component could also be responsible for rendering either `ConfirmMsgSign` or `ConfirmTransaction`, but I like this style better because it _should_ re-render less (less props will change) and seems like a nice separation of concerns.

Reply:
Nice :heart: 

> Alternatively, this added component could also be responsible for rendering either `ConfirmMsgSign` or `ConfirmTransaction`, but I like this style better because it should re-render less (less props will change) and seems like a nice separation of concerns.

Agreed, I like it too :-)